IntegrationTests/tests_01_allocation_counters/test_01_resources/test_stream_teardown_100_concurrent.swift

```diff
@@ -158,29 +156,68 @@ fileprivate class SendGoawayHandler: ChannelInboundHandler {
     )
 
     private let expectedStreams: Int
-    private var seenStreams: Int
+    private let seenStreams = ManagedAtomic<Int>(0)
 
     init(expectedStreams: Int) {
         self.expectedStreams = expectedStreams
-        self.seenStreams = 0
     }
 
     func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
         if event is NIOHTTP2StreamCreatedEvent {
-            self.seenStreams += 1
-            if self.seenStreams == self.expectedStreams {
+            self.seenStreams.wrappingIncrement(ordering: .sequentiallyConsistent)
+            if seenStreams.load(ordering: .sequentiallyConsistent) == self.expectedStreams {
                 // Send a GOAWAY to tear all streams down.
                 context.writeAndFlush(self.wrapOutboundOut(SendGoawayHandler.goawayFrame), promise: nil)
             }
         }
     }
 }
 
+fileprivate struct SendGoawayDelegate: @unchecked Sendable {
+    private static let goawayFrame: HTTP2Frame = HTTP2Frame(
+        streamID: .rootStream, payload: .goAway(lastStreamID: .rootStream, errorCode: .enhanceYourCalm, opaqueData: nil)
+    )
+
+    private let expectedStreams: Int
+    private let seenStreams = ManagedAtomic<Int>(0)
+
+    init(expectedStreams: Int) {
+        self.expectedStreams = expectedStreams
+    }
+}
+
+extension SendGoawayDelegate: NIOHTTP2StreamDelegate {
+    func streamCreated(_ id: HTTP2StreamID, channel: Channel) {
+        self.seenStreams.wrappingIncrement(ordering: .sequentiallyConsistent)
+        if seenStreams.load(ordering: .sequentiallyConsistent) == self.expectedStreams {
+            // Send a GOAWAY to tear all streams down.
+            channel.parent!.writeAndFlush(NIOAny(SendGoawayDelegate.goawayFrame), promise: nil)
+        }
+    }
+
+    func streamClosed(_ id: HTTP2StreamID, channel: Channel) {
+        // do nothing
+    }
+}
+
 func run(identifier: String) {
     var benchmark = StreamTeardownBenchmark(concurrentStreams: 100)
 
     measure(identifier: identifier) {
-        return try! benchmark.run()
+        return try! benchmark.run() { channel, concurrentStreams in
+            _ = try channel.configureHTTP2Pipeline(mode: .server) { streamChannel -> EventLoopFuture<Void> in
+                return streamChannel.pipeline.addHandler(DoNothingServer())
+            }.wait()
+            try channel.pipeline.addHandler(SendGoawayHandler(expectedStreams: concurrentStreams)).wait()
+        }
+    }
+
+    measure(identifier: identifier + "_inline") {
+        return try! benchmark.run() { channel, concurrentStreams in
+            _ = try channel.configureHTTP2Pipeline(mode: .server, connectionConfiguration: .init(), streamConfiguration: .init(), streamDelegate: SendGoawayDelegate(expectedStreams: concurrentStreams)) { streamChannel -> EventLoopFuture<Void> in
```

Comment:
Please try to keep lines to a reasonable length (~120-160), it improves readability.